salt/modules/file.py

```diff
@@ -4315,7 +4315,9 @@ def source_list(source, source_hash, saltenv):
                         ret = (single_src, single_hash)
                         break
                 elif proto.startswith("http") or proto == "ftp":
-                    query_res = salt.utils.http.query(single_src, method="HEAD")
+                    query_res = salt.utils.http.query(
+                        single_src, method="HEAD", decode_body=False
```

Comment:
Thinking this one a bit more thoroughly, I imagine there could be a source that would require the body to be decoded?

Reply:
This is simply checking that the file exists on the remote end there and it's a valid source, there shouldn't be any resource we need to decode the body.